### PR TITLE
Added the ability to install bower and npm on module install

### PR DESCRIPTION
**Pull request by daviesgeek**
Test PR for https://github.com/pyrocms/pyrocms/issues/4027

### CODE REVIEW

**Comment by daviesgeek** [comment]
@RyanThompson here's what I have so far. I'm not sure if this is too naive of a solution?

**Comment by daviesgeek** [comment]
A couple of missing things:

This also needs to be implemented for extensions, not just modules. I didn't add this for extensions, as I wanted to see if this was on the right path.

Field types are not installed, and thus, I'm not really sure exactly how to get the dependencies installed. I'll think about this and see if I can come up with a good plan for it.

File: src/Addon/Module/Command/InstallModule.php [hunk 1]
```diff
@@ -69,6 +69,10 @@ public function handle(
 
         $modules->install($this->module);
 
+        $modules->install($this->module);
```
Comment (RyanThompson) [inline src/Addon/Module/Command/InstallModule.php]: This is duplicated?

**Comment by RyanThompson** [comment]
Also does this install it within the addon itself or in the project root?

**Comment by daviesgeek** [comment]
It installs within the addon; not the root. That way you can reference it relative to the addon's root within your views or whatever needs it.

**Comment by daviesgeek** [comment]
Added support for using yarn if there's a `yarn.lock` file in the root of the project, and yarn is installed.

**Comment by RyanThompson** [comment]
Sorry man - Ima sit on this for a little while.

**Pull request closed without merging**